pallets/messages/src/types.rs

```diff
@@ -40,3 +40,43 @@ where
 		}
 	}
 }
+
+/// Newtype for CIDv2 addresses. We can change the inner type as we understand
+/// more about how to structure IPFS addressable content.
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct CIDv2(Vec<u8>);
```

Comment:
I'm not totally sure this should be Vec<u8> ?
I thought this is just sha256 hash with I think cos a constant size. If so it might make sense to use an array with constant size.

Reply:
It seems like CID size is depend on the Hash function being used, so if we don't specify the hash type then it might makes sense to have vec<u8> to have more flexibility for different hash functions but we loose the checks to make sure if a provided CID is a valid one by checking the size.

Reply:
It looks like [RFC2392](https://datatracker.ietf.org/doc/html/rfc2392) is the CIDv2 specification. It's from 1998 and I don't think it ever was accepted.  Let's consider it YAGNI and kill this code unless there is a CIDv2 AND we decide to support it.  Someone correct me if I'm wrong (in which case my search skills are terrible).  But definitely IPFS has nothing about it [in their docs](https://docs.ipfs.io/concepts/content-addressing/).